server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/Repository.java

```diff
@@ -388,4 +435,46 @@ default Author author() {
     default <T> CompletableFuture<QueryResult<T>> watch(Revision lastKnownRevision, Query<T> query) {
         return RepositoryUtil.watch(this, lastKnownRevision, query);
     }
+
+    /**
+     * A wrapper class which contains absolute {@code from} and {@code to} {@link Revision}s.
+     * This class is created by {@link #normalizeFromToRevision(Revision, Revision)}
+     * or {@link #normalizeFromToRevision(Revision, Revision, boolean)}.
+     */
+    final class NormalizedFromToRevision {
```

Comment:
How about:

- extracting to the top level
- renaming to `RevisionRange`
- moving to the `common` package for future use
- removing `swapped` unless it's really necessary
- allowing to hold relative revisions as well so that it can be used for more use cases
- adding `isRelative()` which would be like `return from.isRelative() || to.isRelative()`

..?